operator/v1/types_klusterlet.go

```diff
@@ -172,6 +172,28 @@ type RegistrationConfiguration struct {
 	// But if the user updates the content of a failed bootstrapkubeconfig, the "failed" mark will be cleared.
 	// +optional
 	BootstrapKubeConfigs BootstrapKubeConfigs `json:"bootstrapKubeConfigs,omitempty"`
+
+	// This provides driver details required to register with hub
+	// +optional
+	RegistrationDriver RegistrationDriver `json:"registrationDriver,omitempty"`
+}
+
+type RegistrationDriver struct {
+	// Type of the authentication used by managedcluster to register as well as pull work from hub. Possible values are csr and awsirsa.
+	// +kubebuilder:default:=csr
+	// +kubebuilder:validation:Enum=csr;awsirsa
+	AuthType string `json:"authType"`
+
+	// Contain the details required for registering with eks hub cluster using IAM roles for service account.
```

Comment:
```suggestion
	// Contain the details required for registering with hub cluster (ie: an EKS cluster) using AWS IAM roles for service account.
```
I think it's better to be consistent and use EKS in upper case in all the places.